### PR TITLE
NotificationService can notify on release date

### PR DESCRIPTION
**Pull request by hedgehog1833**

### CODE REVIEW

**Comment by hedgehog1833** [comment]
[Make Notifications configurable](https://trello.com/c/hlDkMGvu/255-make-notifications-configurable)

File: support/src/main/java/rocks/metaldetector/support/Endpoints.java [hunk 1]
```diff
@@ -70,6 +70,9 @@
     public static final String USERS      = "/rest/v1/users";
     public static final String NOTIFY     = "/rest/v1/notify";
 
+    public static final String FREQUENCY = "/frequency";
+    public static final String RELEASE_DATE = "/release-date";
+
```
Comment (DanielW1987) [inline support/src/main/java/rocks/metaldetector/support/Endpoints.java:75]: here we have the same issue as in the update email pr...i like to have complete rest paths.

File: webapp/src/main/java/rocks/metaldetector/service/notification/NotificationServiceImpl.java [hunk 2]
```diff
@@ -50,6 +51,20 @@ public void notifyOnFrequency() {
         .forEach(config -> frequencyNotification(config, releaseContainer.get(config.getFrequencyInWeeks())));
   }
 
+  @Override
+//  @Scheduled(cron = "0 0 7 * * *")
```
Comment (DanielW1987) [inline webapp/src/main/java/rocks/metaldetector/service/notification/NotificationServiceImpl.java:55]: I don't understand the cron expression. We need to toggle the job once per day, right?
Is it once per day at 7 o'clock?

File: webapp/src/main/resources/templates/email/todays-releases-email.html [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+<head>
+    <title>Today's metal album releases</title>
+</head>
+<body>
+<section layout:fragment="content">
+    <p>
+        Hi <span th:text="${username}">username</span>
+    </p>
+    <p>
+        Here are your favorite band's today's releases:
```
Comment (DanielW1987) [inline webapp/src/main/resources/templates/email/todays-releases-email.html:15]: Looks strange with the two `'`...maybe something like _These are today's releases of your followed bands_

File: webapp/src/test/java/rocks/metaldetector/service/notification/NotificationServiceImplTest.java [hunk 4]
```diff
@@ -311,6 +312,194 @@ void test_inactive_users_not_notified() {
...
+      // when
+      underTest.notifyOnReleaseDate();
+
+      // then
+      verifyNoInteractions(emailService);
+    }
+
+    @Test
+    @DisplayName("NotificationConfigRepository is called on notification for all to get configs")
+    void notify_all_calls_user_service() {
```
Comment (DanielW1987) [inline webapp/src/test/java/rocks/metaldetector/service/notification/NotificationServiceImplTest.java:433]: test method name is wrong

File: webapp/src/test/java/rocks/metaldetector/service/notification/NotificationServiceImplTest.java [hunk 5]
```diff
@@ -311,6 +312,194 @@ void test_inactive_users_not_notified() {
...
+      // when
+      underTest.notifyOnReleaseDate();
+
+      // then
+      verify(notificationConfigRepository).findAll();
+    }
+
+    @Test
+    @DisplayName("Mail- and FollowArtistServices are called times the number of active users on notification for all users")
+    void notify_all_calls_all_services_for_each_user() {
```
Comment (DanielW1987) [inline webapp/src/test/java/rocks/metaldetector/service/notification/NotificationServiceImplTest.java:443]: test method name is wrong

File: webapp/src/main/java/rocks/metaldetector/service/notification/NotificationServiceImpl.java (hunk 2 above)
Comment (hedgehog1833) [reply webapp/src/main/java/rocks/metaldetector/service/notification/NotificationServiceImpl.java]: Yes. Spring - in contrast to all online cron calculators I found - has six didgets: second, minute, hour, day, month, weekday. So this one triggers every day at 7 o'clock.

File: webapp/src/main/resources/templates/email/todays-releases-email.html (hunk 3 above)
Comment (hedgehog1833) [reply webapp/src/main/resources/templates/email/todays-releases-email.html]: I don't like the 'of', maybe that way?

File: webapp/src/main/java/rocks/metaldetector/service/notification/NotificationServiceImpl.java [hunk 6]
```diff
@@ -52,7 +53,7 @@ public void notifyOnFrequency() {
   }
 
   @Override
-//  @Scheduled(cron = "0 0 7 * * *")
+  @Scheduled(cron = "0 0 7 * * *")
```
Comment (DanielW1987) [inline webapp/src/main/java/rocks/metaldetector/service/notification/NotificationServiceImpl.java:56]: Don't we want to enable this only when it is really round and we have tested it well at least on PV?

**Review by DanielW1987 — APPROVED** [review]
Don't forget to comment the scheduling

**Pull request merged**